ironfish-zkp/src/circuits/spend.rs

```diff
@@ -0,0 +1,326 @@
+use bellman::{Circuit, ConstraintSystem, SynthesisError};
+use ff::PrimeField;
+
+use super::util::expose_value_commitment;
+use bellman::gadgets::blake2s;
+use bellman::gadgets::boolean;
+use bellman::gadgets::multipack;
+use bellman::gadgets::num;
+use bellman::gadgets::Assignment;
+use zcash_primitives::{
+    constants::CRH_IVK_PERSONALIZATION,
+    constants::PRF_NF_PERSONALIZATION,
+    sapling::{PaymentAddress, ProofGenerationKey, ValueCommitment},
+};
+use zcash_proofs::{
+    circuit::{
+        ecc::{self},
+        pedersen_hash,
+    },
+    constants::{
+        NOTE_COMMITMENT_RANDOMNESS_GENERATOR, NULLIFIER_POSITION_GENERATOR,
+        PROOF_GENERATION_KEY_GENERATOR, SPENDING_KEY_GENERATOR,
+    },
+};
+
```

Comment:
Is this code a fork of the ZCash definition? Would be nice to include a comment breadcrumb link if it is.

Reply:
Yeah, but they wont stay the same as ZCash class.  I will modify these code in my next PR.